src/agoradatatools/process.py

```diff
@@ -16,59 +16,64 @@
+    ):
+        if not function_info:
+            logger.warning(
+                f"No custom transformation function provided for dataset {dataset_name}. Skipping."
+            )
         return None
-    if dataset_name == "biodomain_info":
-        return transform.transform_biodomain_info(datasets=datasets)
-    if dataset_name == "genes_biodomains":
-        return transform.transform_genes_biodomains(datasets=datasets)
-    if dataset_name == "overall_scores":
-        df = datasets["overall_scores"]
-        return transform.transform_overall_scores(df=df)
-    if dataset_name == "distribution_data":
-        return transform.transform_distribution_data(
-            datasets=datasets,
-            overall_max_score=dataset_obj["custom_transformations"][
-                "overall_max_score"
-            ],
-            genetics_max_score=dataset_obj["custom_transformations"][
-                "genetics_max_score"
-            ],
-            omics_max_score=dataset_obj["custom_transformations"]["omics_max_score"],
-        )
-    if dataset_name == "team_info":
-        return transform.transform_team_info(datasets=datasets)
-    if dataset_name == "rnaseq_differential_expression":
-        return transform.transform_rnaseq_differential_expression(datasets=datasets)
-    if dataset_name == "gene_info":
-        return transform.transform_gene_info(
-            datasets=datasets,
-            adjusted_p_value_threshold=dataset_obj["custom_transformations"][
-                "adjusted_p_value_threshold"
-            ],
-            protein_level_threshold=dataset_obj["custom_transformations"][
-                "protein_level_threshold"
-            ],
-        )
-    if dataset_name == "rna_distribution_data":
-        return transform.transform_rna_distribution_data(datasets=datasets)
-    if dataset_name == "proteomics_distribution_data":
-        return transform.transform_proteomics_distribution_data(datasets=datasets)
-    if dataset_name in ["proteomics", "proteomics_tmt", "proteomics_srm"]:
-        df = datasets[dataset_name]
-        return transform.transform_proteomics(df=df)
-    if dataset_name in ["biomarkers", "pathology"]:
-        return transform.immunohisto_transform(
-            datasets=datasets, dataset_name=dataset_name
-        )
-    if dataset_name == "model_details":
-        return transform.transform_model_details(datasets=datasets)
+    if isinstance(function_info, str):
+        try:
+            retrieved_function = getattr(transform, function_info)
+        except AttributeError:
+            raise AttributeError(
+                f"Function {function_info} not found in the transform module. Please provide the correct function name."
+            )
+        config_defined_params = {}
     else:
-        return None
+        # Retrieve the function name and its parameters
+        # Assumes a single function in the dict
+        function_name, config_defined_params = next(iter(function_info.items()))
```

Comment:
This just grabs the first key and value in a dictionary, right? Should we enforce that functions configured this way can only have one key/value pair? I think datasets can only have one custom transform.

Reply:
@jaclynbeck-sage @beatrizsaldana Can you confirm whether or not we will only ever have at most one custom transform per dataset?

Reply:
@BWMac can I keep this part as it is then? It sounds like we could assume datasets have one custom transform for now.

Reply:
It does sound like it's fine just to grab the first key/value pair from this dictionary. I wonder if it would be clearer if that's what this is doing if we did something like 
```
function_name = list(function_info.keys())[0]
config_defined_params = function_info[function_name]
```
instead, but I think it is fine as is. My other question is whether we should enforce that there is only one custom transform with an exception, or at least throw a warning here if there are more than one that only the first one is being applied.

Reply:
Yeah, I think we could print a warning if there's more than one custom transformation.

Reply:
Agree, it's fine to assume only one custom transform allowed for now, and print a warning if someone puts multiple. We can always update down the road if needed.